### PR TITLE
Add Fuzzy matching option and alias string stripping

### PR DESCRIPTION
**Pull request by dianaabishop**
# Fuzzy Matching
This adds the option to do fuzzy matching on the alias counts. Not perfect because it just retains the alias of the first one it matches on, but a decent start, I think!

The script arg defaults to 100% match but you can pass a --fuzzy-threshold argument with lower numbers if you want to include fuzzy matching.

Good news! this is actually fast - it seemed slow because I was printing the output, but if you don't have any logging it will process thousands of alerts within seconds.

# Remove Numbers
I also added a --remove-numbers flag to be used with fuzzy matching. It strips the numbers out of the two aliases before comparing them. Added this because the shorter the string the lower you have to set the Levenshtein distance to get a match (even if the alerts are literally one number apart). This might be helpful just for bitly alerts but...

# Alias Strip List
Two things to admit about this:
1) I meant to do this on another branch but then I accidentally did it on this branch and then left it
2) Not my strongest naming game - lots of "stripping" even though incidentally I don't use strip() at all - please tell me if you can think of a better one

Basically what this does is allow you to pass a header-less one-column csv with a list of strings you want to strip before doing the matching (I wanted it to ditch server names - could also be used for data centers, etc). This also probably doesn't do great things for the O Notation, but...use at your own risk.

# Requirements
Also I added some dependancies so I included a requirements.txt

### CODE REVIEW

File: count.py [hunk 1]
```diff
@@ -40,10 +52,20 @@ def count(file, column, limit, interval, match, update_minutes, outfile):
...
+                count_map[row[index]] = {"count": (current_count + 1), "fuzzy": False}
+
+        # if the fuzzy threshold is set to 100% match, we can skip this
+        # O Notation murdering step
+        if fuzzy_thresh < 100:
+            # Get a list of the keys in the count_map
+            # Then see if any are a fuzzy match
+            count_map = self.count_fuzzy_matches(count_map, fuzzy_thresh, remove_numbers)
+
+        alert_list = sorted(count_map.items(),
```
Comment (tlugger) [inline count.py:68]: It looks like we miss out on the sorted response with this change, is there a way to get that back?

File: count.py [hunk 2]
```diff
@@ -40,10 +52,20 @@ def count(file, column, limit, interval, match, update_minutes, outfile):
...
                     if dtime.hour < int(interval[0]) or dtime.hour > int(interval[1]):
                         continue
-                count_map[row[index]] = count_map.get(row[index], 0) + 1
 
-        alert_list = sorted(count_map.items(),
-                            key = lambda kv:(kv[1], kv[0]), 
+                current_count = count_map.get(row[index], 0)
+                if current_count != 0:
+                    current_count = current_count.get('count')
+                count_map[row[index]] = {"count": (current_count + 1), "fuzzy": False}
```
Comment (tlugger) [inline count.py:59]: If we don't use fuzzy matching and this dict key is set to False by default could we keep the same `column: count` formatting?

**Review by tlugger — COMMENTED** [review]
Couple thoughts but I like this added functionality

File: count.py (hunk 2 above)
Comment (dianaabishop) [reply count.py:59]: sure thing 👍

File: count.py (hunk 1 above)
Comment (dianaabishop) [reply count.py:68]: believe so - lemme fuss around; I think I ditched this early on but maybe not for the right reason

File: count.py (hunk 2 above)
Comment (dianaabishop) [reply count.py:59]: I decided to keep this here, but I remove it before writing to csv; otherwise it got a little messy if there was a fuzzy matching request...

**Review by tlugger — APPROVED** [review]
The fuzzy matching is awesome!
